src/databricks/sql/thrift_backend.py

```diff
@@ -260,19 +265,40 @@ def make_request(self, method, request):
         def get_elapsed():
             return time.time() - t0
 
+        def make_bounded_delay(attempt: int, delay: int):
+            """Bound delay (seconds) by [min_delay*1.5^(attempt-1), max_delay]"""
+            delay = max(delay, self._retry_delay_min * math.pow(1.5, attempt - 1))
+            delay = min(delay, self._retry_delay_max)
+            return delay
+
         def extract_retry_delay(attempt):
-            # encapsulate retry checks, returns None || delay-in-secs
-            # Retry IFF 429/503 code + Retry-After header set
+            """Returns Retry-After header contents IFF 429/503 code + Retry-After header set"""
             http_code = getattr(self._transport, "code", None)
             retry_after = getattr(self._transport, "headers", {}).get("Retry-After")
             if http_code in [429, 503] and retry_after:
-                # bound delay (seconds) by [min_delay*1.5^(attempt-1), max_delay]
-                delay = int(retry_after)
-                delay = max(delay, self._retry_delay_min * math.pow(1.5, attempt - 1))
-                delay = min(delay, self._retry_delay_max)
-                return delay
+                return make_bounded_delay(attempt, int(retry_after))
             return None
 
+        def get_retry_delay(attempt):
+            """Encapsulate retry checks. Returns None || delay-in-secs
+
+            Retry if
+              - 429/503 code + Retry-After header set
+              - method is GetOperationStatus
+
+            Retry-After header is always used if present. If not present but method is GetOperationStatus
+            the DEFAULT_RETRY_DELAY_AFTER is used. This catches connection timeouts during GetOperationStatus.
+            """
+
+            header_delay = extract_retry_delay(attempt)
+            if (
+                method.__name__ == TCLIServiceClient.GetOperationStatus.__name__
```

Comment:
nit: perhaps it's slightly cleaner to check on the type of the request instead of reflecting on the method name?

Reply:
I wondered about this but don't see an obvious way to assert it. The type of operation is not reflected in Python's type system. 

```python
>>> type(TCLIServiceClient.GetOperationStatus)
<class 'function'>

>>> type(method)
<class 'method'>

>>> isinstance(method, TCLIServiceClient.GetOperationStatus)
False
```

I don't know a cleaner way to check for this condition.